src/sdi-snapd-client-factory.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include <snapd-glib/snapd-glib.h>
+
+void sdi_snapd_client_factory_set_custom_path(gchar *path);
+SnapdClient *sdi_snapd_client_factory_new_snapd_client(void);
```

Comment:
nit: Normally have an empty line between functions.

Reply:
Good point. I'll try to add it to clang-format.